packages/connect-core/src/utils/app.ts

```diff
@@ -1,11 +1,32 @@
 import { utils as ethersUtils } from 'ethers'
 
-import { AppMethod } from '../types'
+import { Abi, AppMethod } from '../types'
 import App from '../entities/App'
 
 export const apmAppId = (appName: string): string =>
   ethersUtils.namehash(`${appName}.aragonpm.eth`)
 
+const checkSignature = (signature: string, abi: Abi): string => {
+  const regex = signature.match(/(.*)\((.*)\)/m)!
+
+  const name = regex[1]
+  const params = regex[2].split(',')
+
+  // If a single ABI node is found with function name and same number of parameters,
+  // generate the signature from ABI. Otherwise, use the one from artifact.
+  const functionAbis = abi
+    .filter((node) => node.name === name)
+    .filter((node) => node.inputs.length === params.length)
+
+  if (functionAbis.length === 1) {
+    return `${functionAbis[0].name}(${functionAbis[0].inputs.map(
+      (input) => input.type
+    )})`
```

Comment:
What would you think of making the comma explicit? It took me a while to remember it was the default of `Array#toString()` haha
 
```suggestion
    return `${functionAbis[0].name}(${functionAbis[0].inputs.map(
      (input) => input.type
    ).join(',')})`
```